stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,259 @@
+        nodes = ""
+
+
+        ###################
+        # parse json file #
+        ###################
+        # load JSON object as dictionary
+        data = json.load(f)
+
+        header = data["header"]
+
+        input_folder = header["input_folder"]
+
+        # make the output folder
+        output_folder = header["output_folder"]
+        try:
+            os.mkdir(f"./{output_folder}")
+        # delete the folder and remake it if it does exist
+        except OSError as error:
+            print(f"NOTICE: {output_folder} folder already exists")
+            shutil.rmtree(output_folder) # delete folder and contents
+            os.mkdir(f"./{output_folder}")
+
+        # load provided textures
+        Textures.add_textures("textures.json")
+        # (optional) load extra textures 
+        if "extra_textures" in header: 
+            Textures.add_textures(header["extra_textures"])
+        
+        # initialize root node
+        nodes += f'[node name="{output_folder}" type="Spatial"]\n'
+
+        # (optional) set scale of scene
+        if "scale" in header:
+            scale = header["scale"]
+            nodes += f'transform = Transform( {scale}, 0, 0, 0, {scale}, 0, 0, 0, {scale}, 0, 0, 0 )\n'
+        nodes += "\n" # add extra new line to prepare for next node
+
+
+        ##################################
+        # iterate through all .stl files #
+        ##################################
+        meshes = data["meshes"]
+
+        # keep track of textures so we don't generate them more than once
+        texture_index = dict()
+
+        for m in meshes:
+            ####################
+            # generate texture #
+            ####################
+            texture = m["texture"]
+
+            # don't generate textures more than once
+            if texture not in texture_index:
+                ext_res_addon, ext_resource_id, sub_res_addon = generate_texture(texture, ext_resource_id, sub_resource_id)
+                ext_resource += ext_res_addon
+                sub_resource += f"{sub_res_addon}\n"
+                    
+                # add the sub resource's index to texture_index so we can use it later 
+                texture_index[texture] = sub_resource_id
+                sub_resource_id += 1 
+
+
+            #####################
+            # generate obj mesh #
+            #####################
+            stl_file = m["stl_file"]
+            stl_file_extless, _ = os.path.splitext(stl_file)
+            stl_file_location = f"{input_folder}/{stl_file}"
+
+            # output filename without a file extension
+            output_prefix = output_folder + "/" + stl_file_extless
+            
+            # Generate .obj file
+            if "limited_dissolve" in m:
+                generate_obj_file(stl_file_location, output_prefix, m["uv_map"], limited_dissolve=False)
+            else:
+                generate_obj_file(stl_file_location, output_prefix, m["uv_map"])
```

Comment:
Confusing logic choice - the existing of `'limited_dissolve'` means that it's *False*?  Is that correct?
If so, then I suggest:
```suggestion
            generate_obj_file(stl_file_location, output_prefix, m["uv_map"], limited_dissolve=('limited dissolve' not in m))
```

Reply:
Yeah, if `limited_dissolve` exists then we don't run limited dissolve on the mesh. This is because the default is that we do apply limited dissolve.